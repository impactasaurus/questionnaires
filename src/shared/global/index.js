import { injectGlobal } from 'styled-components';
import styledReset from 'styled-reset';

export default () => injectGlobal`
    @import url("https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,300italic,400,400italic");
    ${styledReset}

    * {
        box-sizing: border-box;
    }
`;