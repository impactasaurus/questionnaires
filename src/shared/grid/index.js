import R from './Row';
import C from './Column';

export const Row = R;
export const Column = C;