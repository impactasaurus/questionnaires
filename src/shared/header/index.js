import Header from './Header';
import Sub from './Subheader';

export const Subheader = Sub;

export default Header;