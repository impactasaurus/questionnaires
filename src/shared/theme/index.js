import rawTypeset from './typeset';
import rawColours from './colours';
import rawSpacings from './spacings';

export const typeset = rawTypeset
export const colours = rawColours;
export const spacings = rawSpacings;

export const primaryTheme = {
    fg: colours.gray60,
    bg: 'transparent',
    highlight: colours.green50,
    lowlight: colours.white,
    gradient: colours.primaryGradient,
};

export const secondaryTheme = {
    fg: colours.white,
    bg: colours.blue30,
    highlight: colours.white,
    lowlight: 'transparent',
    gradient: colours.primaryGradient,
};