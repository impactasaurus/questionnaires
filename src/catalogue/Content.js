import React, { Component } from 'react';
import gql from 'graphql-tag';
import { Query } from 'react-apollo';
import { Section, Header, Link, Text, Table, Row, Column } from '../shared';
import { TableHeader, TableBody } from '../shared/table';

// TODO: We are loading the whole list of questions here just to know the
// number of questions
export const questionnaireListQuery = gql`
{
    questionnaires(page: 0) {
        questionnaires {
            id
            name
            version
            description
            license
            attribution
            questions {
                id
            }
        }
    }
}`;



export default class Content extends Component {
    render() {
    return (
        <Section>
            <Row>
                <Column>
                    <Header underline>Catalogue</Header>
                </Column>
            </Row>
            <Row>
                <Column>
                    <Query query={questionnaireListQuery}>
                        {({loading, error, data}) =>  {
                            if (loading || error) {
                                return null;
                            }

                            return <Table data={data.questionnaires.questionnaires}>
                                <thead>
                                    <TableHeader>Name</TableHeader>
                                    <TableHeader>Version</TableHeader>
                                    <TableHeader>Description</TableHeader>
                                    <TableHeader>License</TableHeader>
                                    <TableHeader>Attribution</TableHeader>
                                    <TableHeader>Number of questions</TableHeader>
                                </thead>
                                <tbody>
                                    <TableBody>{d => <Link to={`/questionnaire/${d.id}`}>{d.name}</Link>}</TableBody>
                                    <TableBody>{d => <Text>{d.version}</Text>}</TableBody>
                                    <TableBody>{d => <Text>{d.description}</Text>}</TableBody>
                                    <TableBody>{d => <Text>{d.license}</Text>}</TableBody>
                                    <TableBody>{d => <Text>{d.attribution}</Text>}</TableBody>
                                    <TableBody>{d => <Text>{d.questions.length}</Text>}</TableBody>
                                </tbody>
                            </Table>
                        }}
                    </Query>
                </Column>
            </Row>
        </Section>);
    }
}
