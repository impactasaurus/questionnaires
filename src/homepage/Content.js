import React from 'react';
import { Text, Section, Button, Header, Row, Column } from '../shared';

export default () => (
    <Section>
        <Row>
            <Column>
                <Header underline>Getting Started</Header>
            </Column>
        </Row>
        <Row>
            <Column>
              <Text>As you look to measure and demostrate your social impact, 
                    it is important to make sure that you are asking the right questions to show the right value. 
                    Using established questionnaires known for their accuracy increases your faith in the results, 
                    and your trustees, investors and donors faith in you.
                </Text>
            </Column>
        </Row>
        <Row>
            <Column>
                <Button href="/catalogue" inverted>Find a questionnaire</Button>
            </Column>
        </Row>
    </Section>
);