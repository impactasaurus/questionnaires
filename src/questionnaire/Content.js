import React, { Component } from 'react';
import { Section, Header, Text, Table, Row, Column, List, ListItem } from '../shared';
import { TableHeader, TableBody } from '../shared/table';

const uppercaseFirstLetter = str => {
    if (!str || !str.replace) {
        return str;
    }

    return str.replace(/^(.)(.*)$/, (_, firstChar, rest) => `${firstChar.toUpperCase()}${rest}`);
}

export default class Content extends Component {
    renderScoring = question => {
        if (question.__typename !== 'LikertQuestion') {
            return null;
        }

        return <List>
            {question.scale.filter(s => s.label).map(s => <ListItem key={s.value}>
                <Text>{s.label} ({s.value})</Text>
            </ListItem>)}
        </List>
    }

    renderScoringsAggregations = () => {
        const { scorings, questions } = this.props.data;

        if (!scorings || !scorings.length) {
            return null;
        }

        return <Row pad>
            <Column centered>
                <Table data={scorings}>
                    <thead>
                        <TableHeader>Scoring</TableHeader>
                        <TableHeader>Name</TableHeader>
                        <TableHeader>Aggregation</TableHeader>
                        <TableHeader>Questions</TableHeader>
                    </thead>
                    <tbody>
                        <TableBody>{() => null}</TableBody>
                        <TableBody>{d => <Text>{d.name}</Text>}</TableBody>
                        <TableBody>{d => <Text>{uppercaseFirstLetter(d.aggregation)}</Text>}</TableBody>
                        <TableBody>{d => <Text>{d.questions
                                    .map(q => questions.findIndex(({ id }) => id === q) + 1)
                                    .join(', ')}</Text>}</TableBody>
                    </tbody>
                </Table>
            </Column>
        </Row>
    }

    render() {
        if(!this.props.data) {
            return null;
        }

        const {
            name,
            description,
            questions,
        } = this.props.data;

        return <Section>
            <Row>
                <Column>
                    <Header underline>{name}</Header>
                </Column>
            </Row>
            <Row>
                <Column>
                    <Text>{description}</Text>
                </Column>
            </Row>
            {this.renderScoringsAggregations()}
            <Row>
                <Column>
                    {<Table data={questions}>
                        <thead>
                            <TableHeader />
                            <TableHeader>Question</TableHeader>
                            <TableHeader>Scoring</TableHeader>
                        </thead>
                        <tbody>
                            <TableBody>{q => <Text>{questions.findIndex(({ id }) => id === q.id) + 1}</Text>}</TableBody>
                            <TableBody>{q => <Text>{q.question}</Text>}</TableBody>
                            <TableBody>{q => this.renderScoring(q)}</TableBody>
                        </tbody>
                    </Table>}
                </Column>
            </Row>
        </Section>;
    }
}
