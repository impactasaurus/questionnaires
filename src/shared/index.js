import RawAction from './actions';
import RawButton from './button';
import RawHeader, { Subheader as Subhead } from './header';
import RawHero from './hero';
import RawLink, {BlankLink as BLink } from './link';
import RawSection from './section';
import RawTable, { VerticalTable as VTable, TableHeader as TH, TableBody as TB } from './table';
import RawList, { ListItem as LI } from './list';
import { Row as R, Column as C } from './grid';
import RawText from './text';
import RawLogo from './logo';

export const Action = RawAction;
export const Button = RawButton;
export const Header = RawHeader;
export const Subheader = Subhead;
export const Hero = RawHero;
export const Link = RawLink;
export const BlankLink = BLink;
export const Section = RawSection;
export const Table = RawTable;
export const TableHeader = TH;
export const TableBody = TB;
export const VerticalTable = VTable;
export const List = RawList;
export const ListItem = LI;
export const Row = R;
export const Column = C;
export const Text = RawText;
export const Logo = RawLogo;