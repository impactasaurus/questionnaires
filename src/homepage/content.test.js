import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { primaryTheme } from '../shared/theme';
import Content from './Content';

describe('homepage <Content />', () => {
    it('should match the snapshot', () => {
        const component = renderer.create(
            <ThemeProvider theme={primaryTheme}>
                <Content />
            </ThemeProvider>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
});