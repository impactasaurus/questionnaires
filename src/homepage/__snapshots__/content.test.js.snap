// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage <Content /> should match the snapshot 1`] = `
.c5 {
  background-color: transparent;
  -webkit-transition: background-color 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out;
  padding: 0.75rem 1.5rem;
  border: 1px solid rgb(153,153,153);
  border-radius: 4px;
  color: rgb(153,153,153);
  text-decoration: none;
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.125rem;
  -moz-letter-spacing: 0.125rem;
  -ms-letter-spacing: 0.125rem;
  letter-spacing: 0.125rem;
}

.c5:hover {
  background-color: rgba(255,255,255,0.25);
}

.c3 {
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  color: rgb(153,153,153);
  line-height: 3.375rem;
  font-size: 2.25rem;
  font-weight: 300;
  margin: 0;
}

.c3::after {
  content: '';
  height: 1px;
  display: block;
  width: 8rem;
  margin-top: 0.75rem;
  border-bottom: 1px solid rgb(153,153,153);
}

.c0 {
  padding: 1.5rem 2.25rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-bottom: 1.5rem;
}

.c1 > * {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c1::after {
  content: '';
  display: table;
  clear: both;
}

.c2 {
  width: 100%;
}

.c4 {
  color: rgb(153,153,153);
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  font-size: 1rem;
  font-weight: 300;
  line-height: 1.5rem;
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h1
        className="c3"
      >
        Getting Started
      </h1>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <p
        className="c4"
      >
        As you look to measure and demostrate your social impact, it is important to make sure that you are asking the right questions to show the right value. Using established questionnaires known for their accuracy increases your faith in the results, and your trustees, investors and donors faith in you.
      </p>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <a
        className="c5"
        href="/catalogue"
      >
        Find a questionnaire
      </a>
    </div>
  </div>
</section>
`;
